UdpHosts/GameServer/Data/CharacterInventory.cs

```diff
@@ -368,4 +371,88 @@ private byte GetInventoryTypeByItemType(byte itemType)
 
         return (byte)result;
     }
+
+    public void EquipItemByGUID(int loadoutId, LoadoutSlotType slot, ulong guid)
+    {
+        //var currentItem = _character.CurrentLoadout.SlottedItems[slot]; // SdbId
+        
+        if (guid != 0)
+        {
+            var item = _items[guid];
+            _character.CurrentLoadout.SlottedItems[slot] = item.SdbId;
+        }
+        else
+        {
+            _character.CurrentLoadout.SlottedItems[slot] = 0;
+        }
+        
+        var loadoutItems = _loadouts[(uint)loadoutId].LoadoutConfigs[0].Items; //PvE Loadout
+
+        if (loadoutItems.Any(i => i.SlotIndex == (byte)slot))
+        {
+            for (int i = 0; i < loadoutItems.Length; i++)
+            {
+                if (loadoutItems[i].SlotIndex == (byte) slot)
+                {
+                    loadoutItems[i].ItemGUID = guid;
+                    break;
+                }
+            }
+        }
+        else
+        {
+            _loadouts[(uint)loadoutId].LoadoutConfigs[0].Items = loadoutItems.Append(new LoadoutConfig_Item() { ItemGUID = guid, SlotIndex = (byte)slot }).ToArray();
+        }
+
+        switch (slot)
+        {
+            case LoadoutSlotType.Glider:
+                _character.StaticInfo = _character.StaticInfo with { LoadoutGlider = _items[guid].SdbId };
+                break;
+            case LoadoutSlotType.Vehicle:
+                _character.StaticInfo = _character.StaticInfo with { LoadoutVehicle = _items[guid].SdbId };
+                break;
+        }
```

Comment:
Should instead just call SetStaticInfo so that the ObserverView and BaseController are updated too.
Could consider doing so in ApplyLoadout instead as well.

Reply:
Fixed in this instance